### PR TITLE
feat: remove solana from swapper asset search

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Until Solana is supported in swapper, remove solana assets from search

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/8016

## Risk

> High Risk PRs Require 2 approvals

Low

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

- Ensure solana assets are still shown/searchable outside of swapper
- Ensure solana assets are not show/searchable in swapper and all other assets are still there

### Engineering

:point_up:

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

:point_up: 

## Screenshots (if applicable)

image
image
image
image
image
image

### CODE REVIEW

File: src/components/TradeAssetSearch/TradeAssetSearch.tsx [hunk 1]
```diff
@@ -104,7 +106,9 @@ export const TradeAssetSearch: FC<TradeAssetSearchProps> = ({
   const handleSubmit = useCallback((e: FormEvent<unknown>) => e.preventDefault(), [])
 
   const popularAssets = useMemo(() => {
-    const unfilteredPopularAssets = popularAssetsByChainId?.[activeChainId] ?? []
+    const unfilteredPopularAssets = (popularAssetsByChainId?.[activeChainId] ?? []).filter(asset =>
+      isSwapper ? asset.chainId !== KnownChainIds.SolanaMainnet : true,
```
Comment (gomesalexandre) [inline src/components/TradeAssetSearch/TradeAssetSearch.tsx:110]: q: why do we only want this condition for `isSwapper` ? Gut feel is other domains would benefit from the same filtering

**Review by gomesalexandre — COMMENTED** [review]
Tested locally as a cross-test against upstream prod/develop

- Confirmed Solana is gone from the available "to" chains select
- Solana is still present as asset search result in `to` 

https://jam.dev/c/042808ad-0321-4633-8587-96c6349d67b4

image

File: src/components/TradeAssetSearch/TradeAssetSearch.tsx (hunk 1 above)
Comment (kaladinlight) [reply src/components/TradeAssetSearch/TradeAssetSearch.tsx:110]: from dm: That component is for both regular send and receive asset search as well as for swapper. We support Solana for everything except swapper so just added the condition to filter accordingly. Just temp logic realistically

**Comment by kaladinlight** [comment]
> Tested locally as a cross-test against upstream prod/develop
> 
> * Confirmed Solana is gone from the available "to" chains select
> * Solana is still present as asset search result in `to`

Good catch. Fixed in https://github.com/shapeshift/web/pull/8039/commits/fadfaac7002e98994fa28f6cacb96e7616236c26

**Review by gomesalexandre — APPROVED** [review]
second time's a charm https://jam.dev/c/d91d57fe-c4e1-4b67-b32f-266c7b4b83f4

**Pull request merged**